react/src/index.tsx

```diff
@@ -29,43 +39,54 @@ interface SchematicHookOpts {
 
 type UseSchematicFlagOpts = SchematicHookOpts & {
   fallback?: boolean;
-}
+};
 
 const SchematicContext = createContext<SchematicContextProps>({
   flagValues: {},
 });
 
-
 const SchematicProvider: React.FC<SchematicProviderProps> = ({
   children,
   client: providedClient,
-  clientOpts,
   publishableKey,
+  ...clientOpts
 }) => {
-  const [client, setClient] = useState<SchematicJS.Schematic | undefined>();
+  const [client, setClient] = useState<SchematicJS.Schematic>();
   const [flagValues, setFlagValues] = useState<Record<string, boolean>>({});
+  /* eslint-disable-next-line react-hooks/exhaustive-deps */
+  const memoizedClientOpts = useMemo(() => clientOpts, [JSON.stringify(clientOpts)]);
+  const { useWebSocket = true } = clientOpts;
 
   useEffect(() => {
+    let cleanupFunction: (() => void) | undefined;
+
     // If a client was explicitly provided, always use this
     if (providedClient) {
       setClient(providedClient);
-      return providedClient.cleanup;
-    }
-
-    // Otherwise, if a publishable key was provided, create a new client
-    // with the client options
-    if (publishableKey === undefined) {
-      return;
+      cleanupFunction = () => {
+        providedClient.cleanup().catch(error => {
+          console.error("Error during cleanup:", error);
+        });
+      };
+    } else {
+      // Otherwise, if a publishable key was provided, create a new client
+      // with the client options
+      const newClient = new SchematicJS.Schematic(publishableKey, {
```

Comment:
A better way in the future may be to update the client with new options rather than creating a new instance on each option update.

Reply:
yeah that makes sense. I actually am fairly concerned about cases where re-render cycles could cause excess clients to get instantiated. there's definitely more to do to improve this.